src/Handlers.jl

```diff
@@ -117,7 +111,7 @@ function register!(r::Router, method::DataType, scheme, host, path, handler)
     vals = splitsegments(r, handler, segments)
     # return a method to get dispatched to
     #TODO: detect whether defining this method will create ambiguity?
-    @eval $(r.sym)(::$method, ::$scheme, ::$host, $(vals...), args...) = $handler
+    r.func(method, scheme, host, vals...) = handler
```

Comment:
I don't think this is doing what you think it's doing. This is calling `r.func` w/ arguments `method, scheme, host, vals...`, then trying to assign `handler` to that.

Reply:
Btw, if you have any ideas how to fix the original problem, that would be great.

When I add the line

```julia
r = HTTP.Router()
```

to Pages.jl, Julia crashes 🤔